code/modules/events/immovable_rod/immovable_rod.dm

```diff
@@ -290,3 +295,29 @@
 /obj/effect/immovablerod/proc/walk_in_direction(direction)
 	destination_turf = get_edge_target_turf(src, direction)
 	SSmove_manager.move_towards(src, destination_turf)
+
+/**
+ * Rod will push the tram to a landmark if it hits the tram from the front/back
+ * while flying parallel.
+ *
+ * Arguments:
+ * * clong - the object the rod hits (in this case, the tram)
+ */
+/obj/effect/immovablerod/proc/rod_vs_tram_battle(clong)
+	var/turf/tram_turf = get_turf(clong)
```

Comment:
Why don't we just pass the turf into the proc instead of an atom reference we never use?

Reply:
It's actually the turf I didn't need to pass along, it's all done based on the rod.

Reply:
Yeah, I removed the turf and changed it to locate() in src.loc